Makefile

```
@@ -0,0 +1,15 @@
+CC ?= gcc
+PREFIX ?= /usr/local
+CFLAGS += -g
```

```suggestion
CFLAGS += -Wall -Wextra -Wpedantic
```
That will show all the warnings. also we don't want debug builds to be the default, so maybe something like this:
```make
DEBUG := 0

ifneq ($(DEBUG),0)
    CFLAGS += -g
endif
```
that way `DEBUG` can be passed as an argument to `make` to enable debug builds like this: `make DEBUG=1`.

---

PHONY is great, will add

That conditional only works with one Make variant. The provided makefile works for both BSD and GNU make. Want to get rid of debug symbols? Run a strip.

---

> PHONY is great, will add
> 
> That conditional only works with one Make variant. The provided makefile works for both BSD and GNU make. Want to get rid of debug symbols? Run a strip.

Then do you think we should strip it in the install target?

---

Typically strip is called after the build is complete, to allow people to have debug builds if they want. Debian for example expects to need to strip after the build is complete.

---

So making the install target like this is ok:
```make
install:
	strip dollarskip
	install -d $(DESTDIR)$(PREFIX)/bin
	install -m 755 dollarskip $(DESTDIR)$(PREFIX)/bin/\$
```

---

Done.